src/plugins/dashboard/public/dashboard_actions/unlink_from_library_action.tsx

```diff
@@ -41,64 +42,37 @@ export class UnlinkFromLibraryAction implements Action<UnlinkFromLibraryActionCo
     } = pluginServices.getServices());
   }
 
-  public getDisplayName({ embeddable }: UnlinkFromLibraryActionContext) {
-    if (!embeddable.getRoot() || !embeddable.getRoot().isContainer) {
-      throw new IncompatibleActionError();
-    }
+  public getDisplayName({ embeddable }: EmbeddableApiContext) {
+    if (!unlinkActionIsCompatible(embeddable)) throw new IncompatibleActionError();
     return dashboardUnlinkFromLibraryActionStrings.getDisplayName();
   }
 
-  public getIconType({ embeddable }: UnlinkFromLibraryActionContext) {
-    if (!embeddable.getRoot() || !embeddable.getRoot().isContainer) {
-      throw new IncompatibleActionError();
-    }
+  public getIconType({ embeddable }: EmbeddableApiContext) {
+    if (!unlinkActionIsCompatible(embeddable)) throw new IncompatibleActionError();
     return 'folderExclamation';
   }
 
-  public async isCompatible({ embeddable }: UnlinkFromLibraryActionContext) {
-    return Boolean(
-      !isErrorEmbeddable(embeddable) &&
-        embeddable.getInput()?.viewMode !== ViewMode.VIEW &&
-        embeddable.getRoot() &&
-        embeddable.getRoot().isContainer &&
-        embeddable.getRoot().type === DASHBOARD_CONTAINER_TYPE &&
-        isReferenceOrValueEmbeddable(embeddable) &&
-        embeddable.inputIsRefType(embeddable.getInput())
-    );
+  public async isCompatible({ embeddable }: EmbeddableApiContext) {
+    if (!unlinkActionIsCompatible(embeddable)) return false;
+    return embeddable.viewMode.value === 'edit' && (await embeddable.canUnlinkFromLibrary());
   }
 
-  public async execute({ embeddable }: UnlinkFromLibraryActionContext) {
-    if (!isReferenceOrValueEmbeddable(embeddable)) {
-      throw new IncompatibleActionError();
+  public async execute({ embeddable }: EmbeddableApiContext) {
+    if (!unlinkActionIsCompatible(embeddable)) throw new IncompatibleActionError();
+    try {
+      await embeddable.unlinkFromLibrary();
+      const panelTitle = embeddable.panelTitle?.value ?? embeddable.defaultPanelTitle?.value;
+      this.toastsService.addSuccess({
+        title: dashboardUnlinkFromLibraryActionStrings.getSuccessMessage(
+          panelTitle ? `'${panelTitle}'` : ''
+        ),
+        'data-test-subj': 'unlinkPanelSuccess',
+      });
+    } catch (e) {
+      this.toastsService.addDanger({
```

Comment:
How about including title in failure message as well

Reply:
Done